apps/meteor/app/statistics/server/lib/SAUMonitor.ts

```diff
@@ -185,6 +189,26 @@ export class SAUMonitorClass {
 		// TODO missing an action to perform on dangling sessions (for example remove sessions not closed one month ago)
 	}
 
+	private async _getSearchTerm(session: Omit<ISession, '_id' | '_updatedAt' | 'createdAt'>): Promise<string | undefined> {
+		const user = await Users.findOneById<IUser>(session.userId, {
+			projection: {
+				_id: 1,
+				username: 1,
+				name: 1,
+			},
+		});
+
+		if (!user) {
+			return;
+		}
+
+		const searchTerm = `${session.device?.name || ''}${session.device?.type || ''}${session.device?.os.name || ''}${session.sessionId}${
+			user.name || ''
+		}${user.username || ''}`;
```

Comment:
not sure if this is a concern or not, but storing the name and username like this will make it impossible to update in case the user changes his name or username, is that an issue?

let's my name when I registered was `Diego Sampaio`, so the `searchTerm` will have this information.. if later on I change my name to `Didi`, when searching for "didi" I'll find only the new records and if I search for "sampaio" only the old ones will be found.

if this limitation is not an issue, maybe we should just add a notice on the regarding it, so people don't get confused when try to find for something and the results are not exactly what they're expecting.